### PR TITLE
final arch board changes for preview other than model renames

### PR DESCRIPTION
**Pull request by m-nash**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: common/ManagementCoreShared/SharedExtensions.cs [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
-namespace Azure.ResourceManager
+namespace Azure.ResourceManager.Core
 {
     /// <summary>
-    /// Extensions over ResourceIdentifier types.
+    /// helper class
     /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
-    public static class ResourceIdentifierExtensions
+    internal static class SharedExtensions
```
Comment (allenjzhang) [inline common/ManagementCoreShared/SharedExtensions.cs:14]: public to internal? So AppendProviderResource and AppendChildResource are now hidden?

Comment (m-nash) [reply common/ManagementCoreShared/SharedExtensions.cs:14]: hidden but shared, they will be included as shared code in all mgmt plane sdk

Comment (allenjzhang) [reply common/ManagementCoreShared/SharedExtensions.cs:14]: I think these are useful methods for enduser of ResourceIdentifier not just internal logic.

File: common/ManagementCoreShared/SharedExtensions.cs [hunk 2]
```diff
@@ -36,27 +36,79 @@ public static ResourceIdentifier AppendProviderResource(this ResourceIdentifier
...
+        /// </summary>
+        /// <param name="dest"> The destination dictionary in which the content will be replaced. </param>
+        /// <param name="src"> The source dictionary from which the content is copied from. </param>
+        /// <returns> The destination dictionary that has been altered. </returns>
+        public static IDictionary<string, string> ReplaceWith(this IDictionary<string, string> dest, IDictionary<string, string> src)
+        {
+            dest.Clear();
+            foreach (var kv in src)
+            {
+                dest.Add(kv.Key, kv.Value);
```
Comment (archerzz) [inline common/ManagementCoreShared/SharedExtensions.cs:57]: I think we can use this directly.
```suggestion
                dest.Add(kv);
```

File: common/ManagementCoreShared/SharedExtensions.cs (hunk 1 above)
Comment (m-nash) [reply common/ManagementCoreShared/SharedExtensions.cs:14]: yes they did, and we had editor browsable never for the same reason.  This is just a better way to hide it.

I think we can add these kinds of helpers later but for preview we want to keep the api surface as small as possible

Comment (ArcturusZhang) [reply common/ManagementCoreShared/SharedExtensions.cs:14]: Does this mean that the customers are not encouraged to use these APIs?

**Review by ArcturusZhang — APPROVED** [review]
Approved with one question

**Pull request merged**